components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -1530,17 +1482,13 @@ void AdsServiceImpl::CanShowNotificationAds(
 void AdsServiceImpl::CanShowNotificationAdsWhileBrowserIsBackgrounded(
     CanShowNotificationAdsWhileBrowserIsBackgroundedCallback callback) {
   std::move(callback).Run(
-      NotificationHelper::GetInstance()
-          ->CanShowSystemNotificationsWhileBrowserIsBackgrounded());
+      delegate_->CanShowSystemNotificationsWhileBrowserIsBackgrounded());
 }
 
 void AdsServiceImpl::ShowNotificationAd(base::Value::Dict dict) {
```

Comment:
Maybe we can move this function to `AdsServiceDelegate` and pass `ShouldShowCustomNotificationAds()` value as a parameter? Declaration can look something like:
```
virtual void ShowNotificationAd(
                                    bool should_show_custom_notification_ads,
                                    const std::string& id,
                                    const std::u16string& title,
                                    const std::u16string& body);
```

If so, we will get rid of `ui/message_center` dependency in header because the new function will replace following functions in `AdsService::Delegate`:
```
    virtual void Display(const message_center::Notification& notification) = 0;
    virtual void ShowNotificationAd(const std::string& id,
                                    const std::u16string& title,
                                    const std::u16string& body);
```

Reply:
I think there's still quite a bit of refactoring that needs to be done here, but for this first pass my goal is just to remove `check_includes = false`